maintenance/health/servicehealthcheck/background_healthchecker.go

```diff
@@ -0,0 +1,89 @@
+package servicehealthcheck
+
+import (
+	"context"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/pace/bricks/maintenance/log"
+)
+
+// BackgroundHealthCheck is an interface that, when implemented, will run the
+// HealthCheck in the background, and cache the result, that will be fetched by the HealthHandler
+type BackgroundHealthCheck interface {
+	HealthCheck
+	Interval() time.Duration
+}
+
+// BackgroundHealthCheckMaxWait allows to use a custom amount of time to wait before failing the health check
+type BackgroundHealthCheckMaxWait interface {
+	MaxWait() time.Duration
+}
+
+// registerBackgroundHealthCheck will run the HealthCheck in the background at every given interval.
+// The returned backgroundStateHealthChecker is then used to query this cached state when needed.
+func registerBackgroundHealthCheck(name string, bhc BackgroundHealthCheck) *backgroundStateHealthChecker {
+	var bgState ConnectionState
+
+	go func() {
+		s := log.Sink{Silent: true}
+		ctx := log.ContextWithSink(context.Background(), &s)
+
+		var initErr error
+		if initHC, ok := bhc.(Initializable); ok {
+			if initErr = initHC.Init(ctx); initErr != nil {
+				log.Warnf("error initializing background health check %q: %s", name, initErr)
+				bgState.SetErrorState(initErr)
+			}
+		}
+		timer := time.NewTimer(0) // Do first health check instantly
+		for {
+			func() {
+				<-timer.C
```

Comment:
can we move this out of the function, because we do not need to start a func when it just waits. Does not really matter in the grand scheme, but it's a function sitting on the call stack for N minutes when it could only sit there when it is actually needed.